ssz/sedes/base.py

```diff
@@ -130,13 +150,21 @@ def serialize(self, value: TSerializable) -> bytes:
         if not len(value):
             return b''
 
-        pairs = self._get_item_sedes_pairs(value)
+        pairs = self._get_item_sedes_pairs(value)  # slow
         element_sedes = tuple(sedes for element, sedes in pairs)
 
-        fixed_size_section_length = _compute_fixed_size_section_length(element_sedes)
+        has_fixed_size_section_length_cache = hasattr(value, '_fixed_size_section_length_cache')
```

Comment:
This approach is probably fine, though I think you can get rid of this complexity by relegating this to an `abstractmethod` and letting each implementation determine if it wants to cache the result value or not.

That way this stays a one-liner and each sedes can decide if it wishes to cache the value or not.

Reply:
Thanks for the feedback!

I think I have to rework `CompositeSedes.serialize()`, `Vector.serialize()`, `List.serialize()`, and `Cointainer.serialize()` so that we can only check `has_fixed_size_section_length_cache` in the `Cointainer.serialize`.

Since this PR is already too big, will leave it to the next PR.